pytorch_lightning/trainer/supporters.py

```diff
@@ -90,3 +92,60 @@ def accumulate(self, x):
 
     def mean(self):
         return self.total / self.num_values
+
+
+class PredictionCollection(object):
```

Comment:
one more question: can't we basically do this as a TestResult (similar to Train and EvalResults)? I think it's quite confusing, when we have yet another class. Or is this one created and handled automatically?

Reply:
to both your comments - I think we could rethink this whole idea a lot better. this is just a quick workaround to fix numerous issues we've had about this. 😄